frontend/Auth/FinishResetPassword.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import { Button, ButtonKind } from "../Styleguide/Button";
+import { FormField } from "../Styleguide/FormField";
+import { FormFieldLabel } from "../Styleguide/FormFieldLabel";
+import { Input } from "../Styleguide/Input";
+import { RouterProps } from "react-router";
+import { useMutation, useQuery } from "react-query";
+import { flaxFetch } from "../Utils/flaxFetch";
+import { unary } from "lodash-es";
+
+export function FinishResetPassword(props: RouterProps) {
+  const [finishResetPasswordFormData, setFinishResetPasswordFormData] =
+    useState<FinishResetPasswordFormData>({
+      password: "",
+    });
+
+  const paramString = props.history.location.search;
+  let searchParams = new URLSearchParams(paramString);
+  const token = searchParams.get("jwt") || "MISSING TOKEN";
+
+  const queryFunctionhelper = (queryKey) => {
+    const token = queryKey["queryKey"][0];
+    return flaxFetch<object>(
+      `/validate-token/${token}?tokenType=passwordReset`,
+      {
+        method: "GET",
+      }
+    );
+  };
+
+  const info = useQuery(token, queryFunctionhelper);
+
+  const submitMutation = useMutation<
+    void,
+    Error,
+    React.FormEvent<HTMLFormElement>
+  >((evt) => {
+    evt.preventDefault();
+    let requestPromise = flaxFetch<void>(`/send-reset-password-email`, {
+      method: "POST",
+      body: {
+        password: finishResetPasswordFormData.password,
+        token: token,
+      },
+    });
+    return requestPromise;
+  });
+
+  return (
+    <div className="flex justify-center h-screen p-24 sm:pt-80">
+      <form
+        onSubmit={unary(submitMutation.mutate)}
```

Comment:
We need to redirect them to the home page after the password is set. Is this something that you plan on adding in a future PR? Could be done with a `useEffect` that checks whether the submitMutation succeeded or not, which is probably better than just adding it straight into the useMutation itself

Reply:
Yes, this will be added in a future PR.